homeassistant/components/rachio/binary_sensor.py

```diff
@@ -64,16 +60,14 @@ def _async_handle_any_update(self, *args, **kwargs) -> None:
         # For this device
         self._async_handle_update(args, kwargs)
 
-    @abstractmethod
-    def _poll_update(self, data=None) -> bool:
-        """Request the state from the API."""
-
     @abstractmethod
     def _async_handle_update(self, *args, **kwargs) -> None:
         """Handle an update to the state of this sensor."""
 
     async def async_added_to_hass(self):
         """Subscribe to updates."""
+        self._state = self._controller.init_data[KEY_STATUS] == STATUS_ONLINE
```

Comment:
This should probably be in `RachioControllerOnlineBinarySensor`. You can add something like this to make sure the base class is still called:
```
async def async_added_to_hass(self):
   self._state = self._controller.init_data[KEY_STATUS] == STATUS_ONLINE
   await super().async_added_to_hass()
```

Reply:
So just to be clear, leave this method in the base class and add the other to the subclass?

```
async def async_added_to_hass(self):
        """Subscribe to updates."""

        self.async_on_remove(
            async_dispatcher_connect(
                self.hass,
                SIGNAL_RACHIO_CONTROLLER_UPDATE,
                self._async_handle_any_update,
            )
        )
```

Reply:
Yes.  At some point we will probably add something else that uses `RachioControllerBinarySensor` as a base class and it shouldn't know about the online state.

Reply:
👍